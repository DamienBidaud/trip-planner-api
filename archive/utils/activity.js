const getActivity = (id, context, info) =>
  context.prisma.query.activity(
    {
      where: {
        id
      }
    },
    info
  );

module.exports = {
  getActivity
};
