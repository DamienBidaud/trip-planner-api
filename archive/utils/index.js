const location = require("./location");
const activity = require("./activity");
const trip = require("./trip");

module.exports = {
  ...location,
  ...activity,
  ...trip
};
