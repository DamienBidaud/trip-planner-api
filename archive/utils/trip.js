const getTrip = (id, context, info) =>
  context.prisma.query.trip(
    {
      where: {
        id
      }
    },
    info
  );

module.exports = {
  getTrip
};
