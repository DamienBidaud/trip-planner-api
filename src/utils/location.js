const findLocation = async (context, location) => await context.prisma.query.location(
  {
    where: {
      city: location.city,
      country: location.country
    }
  },
  `{
    id
    country
    city
    longitude
    latitude
    type
  }`
);

const addLocation = async (context, location) => await context.prisma.mutation.createLocation({
  data: location
}, `{ id }`);

module.exports = {
  findLocation,
  addLocation
};