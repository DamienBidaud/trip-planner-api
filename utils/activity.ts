import { Context } from "../index.d";

export const getActivity = async (id: number, context: Context) =>
  await context.prisma.activity.findOne({
    where: {
      id,
    },
  });
