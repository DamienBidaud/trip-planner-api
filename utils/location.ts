import { Context } from "../index.d";
import { Location } from "@prisma/client";

export const findLocation = async (
  context: Context,
  location: { city: string; country: string }
) =>
  await context.prisma.location.findMany({
    where: {
      city: location.city,
      country: location.country,
    },
  });

export const addLocation = async (context: Context, location: Location) =>
  await context.prisma.location.create({
    data: location,
  });
