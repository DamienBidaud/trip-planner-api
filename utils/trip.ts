import { Context } from "../index.d";

export const getTrip = async (id: number, context: Context) =>
  await context.prisma.trip.findOne({
    where: {
      id,
    },
  });
