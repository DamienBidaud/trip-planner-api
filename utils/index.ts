import * as location from "./location";
import * as activity from "./activity";
import * as trip from "./trip";

export default {
  ...location,
  ...activity,
  ...trip,
};
