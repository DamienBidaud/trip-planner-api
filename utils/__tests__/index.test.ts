import utils from "../index";

describe("utils", () => {
  it("Should match snapshot", () => {
    expect(utils).toMatchSnapshot();
  });
});
