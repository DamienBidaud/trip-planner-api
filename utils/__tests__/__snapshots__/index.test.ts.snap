// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils Should match snapshot 1`] = `
Object {
  "addLocation": [Function],
  "findLocation": [Function],
  "getActivity": [Function],
  "getTrip": [Function],
}
`;
